### DISCUSSION TITLE
Docker installation of Locast2Plex on Synology NAS

### DISCUSSION THREAD

**Discussion by CTJohnK** [tgorgdotcom/locast2plex#172] [General]
I broke down today and decided to move my L2P instance from my Raspberry Pi (0.4.2 has been working fine) to a Docker container on my NAS.  There were a couple of hiccups since I had never used Docker before, but I have it working now.

If anyone is interested, I've included the steps.

1.	Install Docker from the Package Center.
  a.	The installation will create a folder on your Synology NAS called ‘docker’.
2.	From Control Panel, select the ‘docker’ folder, then ‘Edit’ and make sure at least your User ID has Read/Write permission.
  a.	This is an optional step but makes configuring Locast2Plex easier in our later steps.
3.	From File Station, navigate to the ‘docker’ folder and create a sub-folder (‘L2P-Folder’ for example).
  a.	This is where you will save the ‘config.ini’ file later.
4.	Open the Docker app and click on ‘Registry’.
  a.	In the search field, type ‘locast’ then click ‘Search’.
  b.	Double-click on ‘tgorg/locast2plex’.  (I would recommend selecting the ‘latest’ Tag option and click ‘Select’ to download the required files.)
  c.	You can click ‘Image’ within 'Docker’ to monitor the download progress.
5.	Copy the config_example.ini (https://github.com/tgorgdotcom/locast2plex/blob/main/config_example.ini) and place it in the folder you created earlier under the ‘docker’ folder with the name ‘config.ini’.
6.	Modify then save your config.ini -
  a.	plex_accessible_ip = 127.0.0.1
        i.	This is the ‘loop back’ IP address, you could also use the IP address of your Synology NAS.  **In either case this IP address should be the IP address of the machine running Locast2Plex.**
  b.	Also edit the ‘locast_username’ and ‘locast_password’ to match your Locast account.
  c.	The rest of the settings can be left alone for now.
       i.	I did set ‘tuner_count = 4’ which is the maximum number of concurrent sessions Locast allows.  This will allow you to watch up to 4 shows via Plex.
7.	Once the image has been downloaded, double-click on the image to open the settings.
  a.	Change the ‘Container Name’ if desired and leave the other options alone.
  b.	Click ‘Advanced Settings’
      i.	For now, do not enable ‘auto-restart’
      ii.	On the ‘Volume’ tab, click ‘Add File’ this is where we point L2P to our config.ini file created earlier.
             1.	Find your Docker folder and drill down to the ‘config.ini’ file, select and click ‘Select’.
             2.	For the ‘Mount path’ enter ‘/app/config/config.ini’ and do not click ‘Read-Only’.
      iii.	Over to the ‘Port Settings’ tab.
             1.	Click the ‘+’ and enter ‘6077’ for both the ‘Local Port’ and the ‘Container Port’.
      iv.	Click ‘Apply’.
  c.	Click ‘Next’, then ‘Apply’ again.
8.	Now click on ‘Container’ and if all went well, you should have your Locast2Plex running.
9.	Double-click your running container and the ‘Overview, Process, Log and Terminal’ screen opens.
  a.	Click on ‘Log’ and scroll to the bottom, you should see a message ‘Locast2Plex is now online.’
      i.	If not, scroll up and see where an error might be present.
  b.	Presuming the container is running, on the ‘Overview’ tab, click ‘Stop’.
10.	Once the container has stopped, jump back to Docker/Container, highlight your container, and click ‘Edit’.
  a.	Now enable ‘Enable auto-restart’.
11.	Next, click the ‘Action’ button and select ‘Start’.
12.	Again, jumping back to the ‘Overview, Process, Log, Terminal’ screen, check the Log tab to confirm you have another ‘Locast2Plex is now online.’ Message.
13.	You should be all set to now add your Locast2Plex ‘tuner’ to Plex.

Hope this helps.

**Comment by djsiegel** [comment]
@CTJohnK  - thanks for this.. mines been working well for a while. Have you figured out a way to clear logs from the service running on your NAS? I saw you posted another issue elsewhere about getting the logs of docker itself. I've not been able to have a lot of luck finding where either config or logs really belong on the synology.

**Reply by CTJohnK** [reply]
No I have not and as a matter of fact I had forgotten about it.  I'll have to revisit the issue and if I find anything I'll post it here.
